product_allowed_list/models/product_list_line.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2021 Camptocamp SA
+# @author: Julien Coux <julien.coux@camptocamp.com>
+# @author: Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ProductListLine(models.Model):
+    _name = "product.allowed.list.line"
+    _description = "Product allowed list configuration Line"
+
+    product_allowed_list_id = fields.Many2one(
+        string="Product list configuration",
+        comodel_name="product.allowed.list",
+        required=True,
+    )
+    product_template_id = fields.Many2one(
+        comodel_name="product.template",
+        domain=[("sale_ok", "=", True)],
+        required=True,
+    )
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        domain="[('product_tmpl_id', '=', product_template_id)]",
+    )
+    display_name = fields.Char(compute="_compute_display_name")
+
+    def find_for_product(self, prod, config=None):
```

Comment:
```suggestion
    def _find_for_product(self, prod, config=None):
```
If not exposed through xmlrpc, better to have it as a private method